packages/ui/src/components/va-chip/VaChip.vue

```diff
@@ -35,139 +35,136 @@
-  }
+    } as Record<string, string>
+    const iconSize = computed(() => size[props.size])
 
-  get indexComputed () {
-    return this.disabled ? -1 : 0
-  }
+    const indexComputed = computed(() => props.disabled ? -1 : 0)
 
-  get computedClass () {
-    return {
-      'va-chip--small': this.size === 'small',
-      'va-chip--large': this.size === 'large',
-      'va-chip--square': this.square,
-      'va-chip--disabled': this.disabled,
-    }
-  }
+    const computedClass = computed(() => ({
+      'va-chip--small': props.size === 'small',
+      'va-chip--large': props.size === 'large',
+      'va-chip--square': props.square,
+      'va-chip--disabled': props.disabled,
+    }))
 
-  get shadowStyle () {
-    if (!this.shadow || this.flat || this.outline || this.disabled || this.SetupContext.hasKeyboardFocus) {
-      return
-    }
-    return '0 0.125rem 0.19rem 0 ' + getBoxShadowColor(this.colorComputed)
-  }
+    const shadowStyle = computed(() => {
+      if (!props.shadow || props.flat || props.outline || props.disabled || hasKeyboardFocus.value) {
+        return
+      }
+      return `0 0.125rem 0.19rem 0 ${getBoxShadowColor(colorComputed.value)}`
+    })
 
-  get computedStyle () {
-    const borderColor = this.outline ? this.colorComputed : ''
-    const computedStyle = {
-      color: this.colorComputed,
-      borderColor,
-      background: '',
-      boxShadow: this.shadowStyle,
-    }
+    const computedStyle = computed(() => {
+      const result = {
+        color: colorComputed.value,
+        borderColor: borderColor.value,
+        background: '',
+        boxShadow: shadowStyle.value,
+      }
 
-    if (this.outline || this.flat) {
-      if (this.SetupContext.hasKeyboardFocus) {
-        computedStyle.background = getFocusColor(this.colorComputed)
-      } else if (this.hoverState) {
-        computedStyle.background = getHoverColor(this.colorComputed)
+      if (props.outline || props.flat) {
+        if (hasKeyboardFocus.value) {
+          result.background = getFocusColor(colorComputed.value)
+        } else if (hoverState.value) {
+          result.background = getHoverColor(colorComputed.value)
+        }
+      } else {
+        result.color = getTextColor(colorComputed.value)
+        result.background = colorComputed.value
       }
-    } else {
-      computedStyle.color = getTextColor(this.colorComputed)
-      computedStyle.background = this.colorComputed
-    }
-    return computedStyle
-  }
 
-  updateHoverState (isHover: boolean) {
-    this.hoverState = isHover
-  }
+      return result
+    })
 
-  close () {
-    if (!this.disabled) {
-      this.valueComputed = false
+    const { valueComputed } = useStateful(props, emit)
+    const close = () => {
+      if (!props.disabled) {
+        valueComputed.value = false
+      }
     }
-  }
-}
+
+    return {
+      keyboardFocusListeners,
+      updateHoverState,
+      indexComputed,
+      computedStyle,
+      computedClass,
```

Comment:
What I'm thinking about. Can't we declare computedClass here in return statement? And another things like close, computedStyle and other properties.

This can be better code structure, because you know exactly which properties are required for template and which properties are required for another properties (computes).